### ISSUE TITLE
[Ubuntu 14.04] Microphone doesn't work

### ISSUE THREAD

**Issue by frabcus** [advancingu/XPS13Linux#17]
Seems to need kernel patches:

http://blogs.aerys.in/jeanmarc-leroux/2015/05/13/xps-13-developer-edition-microphone-fix/

**Comment by advancingu** [comment]
Which distribution, distribution release and BIOS version are you on?

**Comment by frabcus** [comment]
Ubuntu 14.04 - the version that came installed on the computer, with the Dell driver repositories and fully up to date.

```
$ uname -a
Linux fox 3.13.0-52-generic #86-Ubuntu SMP Mon May 4 04:32:59 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Bios version is A03, from `dmesg`:

```
[    0.000000] DMI: Dell Inc. XPS 13 9343/0310JH, BIOS A03 03/25/2015
```

**Comment by soren121** [comment]
Barton George noted in his [blog post](http://bartongeorge.net/2015/04/09/4th-gen-dell-xps-13-developer-edition-available/) that the microphone patches were not backported to 14.04...not sure if that also applies to Dell's image, but there you go. I would strongly recommend that you upgrade to 15.04, which does include the microphone patches, as well as a number of other crucial driver patches.

**Comment by frabcus** [comment]
It says - "Currently (4/15) out of the box the HDA microphone will not work, and you will need some packages from the factory shipped image to make it work properly.", and I'm using the factory shipped image (Developer Edition).

**Comment by soren121** [comment]
Odd. Unfortunately, I can't help you with that since Dell chose not to make their custom 14.04 image & packages public.

You really should just upgrade to 15.04. The requisite patches are all in the stock Vivid kernel, no Dell packages necessary.

**Comment by hg8** [comment]
Hello!
I am under Ubuntu 15.04 with kernel 4.1.2 and I can not manage to get the microphone to work.
It is displayed in "Sound settings" but does not work...
Can some one have found a fix for this ? Is this a regression in kernel 4.1.2 ?

**Comment by soren121** [comment]
Yes, it's a regression. You need this patch: https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=831bfdf9520e389357cfeee42a6174a73ce7bdb7

**Comment by hg8** [comment]
Thanks!
Sorry but I am a bit new here... So... What should I do with this link ?

**Comment by soren121** [comment]
Oh, sorry! You need to recompile a **4.1 kernel** with that patch applied. (It's very important that you use a 4.1 series kernel and not an older one.) It's a bit of an involved process on Ubuntu, and I don't use Ubuntu so I don't want to give you bad instructions...try Googling a bit, you should be able to figure things out.

**Comment by hg8** [comment]
Thanks I will give a try. 
Barton George linked two patch in his blog. Is this only for 14.04 ? And the patch you linked is only for 15.04 ?

**Comment by soren121** [comment]
The patches Barton linked to are for kernels 4.0 and older. They were included in kernel 4.1.

There was a last-minute regression in 4.1, however, that broke those patches. The one I linked to fixes that. It's only for 4.1.

Sent from my Windows Phone

-----Original Message-----

**Comment by hg8** [comment]
Good news. Kernel 4.1.3 fixed the problem :)

**Comment by soren121** [comment]
So it did! I was wondering when it would get pulled into Stable.

Sent from my Windows Phone

-----Original Message-----

**Comment by hg8** [comment]
What the.... Yesterday I reinstalled my Ubuntu 15.04 dual booted with Windows 10.
After installing kernel 4.1.3 the microphone still does not work. 
I tried to install kernel 4.1.4 but it did not work either.

I don't understand what is going on, it was working before I reinstall. Anyone encounter the same problem ?

Edit : Now it is working fine again... I have not done anything.
